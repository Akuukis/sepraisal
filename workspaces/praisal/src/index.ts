export * from './Praisal'
export * from './PraisalManager'
export * from './models'
export * from './unzipCachedSbc'
