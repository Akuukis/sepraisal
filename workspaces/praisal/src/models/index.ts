
export * from './Cube.ts'
export * from './Blueprint.ts'
export * from './Component.ts'
export * from './Ingot.ts'
export * from './Ore.ts'
export * from './Orientation.ts'


export * from './Group.ts'
