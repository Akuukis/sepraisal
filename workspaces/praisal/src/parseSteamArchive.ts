import { Stream } from 'stream'
import * as unzip from 'unzip'


export const parseSteamArchive = async (readStream: Stream): Promise<string> => {
    let result: string

    return (new Promise<string>((resolve, reject) => {
        readStream
            .on('error', reject)
            .pipe(unzip.Parse())
                .on('entry', (entry: unzip.Entry) => {
                    try {
                        const fileName = entry.path
                        const chunks: Buffer[] = []

                        entry.on('data', (chunk: Buffer) => {
                            chunks.push(chunk)
                        })

                        entry.on('end', () => {
                            if(fileName.endsWith('.sbc')) {
                                result = Buffer.concat(chunks).toString()
                            }
                        })
                    } catch(err) {
                        reject(err)
                    }
                })
                .on('error', reject)
                .on('close', () => {
                    resolve(result)
                })
    }))

}
