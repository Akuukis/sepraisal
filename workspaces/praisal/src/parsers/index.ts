export * from './parseBlueprint'
export * from './parseBlueprintSbc'
export * from './parseComponentSbc'
export * from './parsePhysicalItemsSbc'
