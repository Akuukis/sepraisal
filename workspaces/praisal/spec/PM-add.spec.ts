import { readFileSync } from 'fs'
import { join } from 'path'

import { PraisalManager } from '../src'


// tslint:disable: no-duplicate-string
const VENDOR_DIR = join(__dirname, '..', 'vendor')


let sepraisal: PraisalManager
beforeEach(() => {
    sepraisal = new PraisalManager()
})

describe('PraisalManager.addOres', () => {
    test('should succeed based on vendor assets', async () => {
        const physicalItemsXml = readFileSync(join(VENDOR_DIR, 'PhysicalItems.sbc')).toString()
        await sepraisal.addOres(physicalItemsXml)
        expect([...sepraisal.ores.entries()][0]).toMatchSnapshot()
        expect([...sepraisal.ores.keys()]).toMatchSnapshot()
        expect(sepraisal.ores.size).toEqual(13)
    })
})
describe('PraisalManager.addIngots', () => {
    test('should succeed based on vendor assets', async () => {
        const materialsXml = readFileSync(join(VENDOR_DIR, 'Blueprints.sbc')).toString()
        const physicalItemsXml = readFileSync(join(VENDOR_DIR, 'PhysicalItems.sbc')).toString()
        await sepraisal.addIngots(physicalItemsXml, materialsXml)
        expect([...sepraisal.ingots.entries()][0]).toMatchSnapshot()
        expect([...sepraisal.ingots.keys()]).toMatchSnapshot()
        expect(sepraisal.ingots.size).toEqual(10)
    })
})
describe('PraisalManager.addComponents', () => {
    test('should succeed based on vendor assets', async () => {
        const componentsXml = readFileSync(join(VENDOR_DIR, 'Components.sbc')).toString()
        const materialsXml = readFileSync(join(VENDOR_DIR, 'Blueprints.sbc')).toString()
        await sepraisal.addComponents(materialsXml, componentsXml)
        expect([...sepraisal.components.entries()][0]).toMatchSnapshot()
        expect([...sepraisal.components.keys()]).toMatchSnapshot()
        expect(sepraisal.components.size).toEqual(23)
    })
})
describe('PraisalManager.addCubes', () => {
    test('should succeed based on vendor assets', async () => {
        const cubeBlocksXmls = [
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Armor.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Automation.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Communications.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Control.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_DecorativePack.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_DecorativePack2.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Doors.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Economy.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Energy.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Extras.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Gravity.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Interiors.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_LCDPanels.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Lights.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Logistics.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Mechanical.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Medical.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Production.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Thrusters.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Tools.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Utility.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Weapons.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Wheels.sbc')).toString(),
            readFileSync(join(VENDOR_DIR, 'CubeBlocks', 'CubeBlocks_Windows.sbc')).toString(),
        ]
        for(const cubeBlocksXml of cubeBlocksXmls) await sepraisal.addCubes(cubeBlocksXml)
        expect([...sepraisal.cubes.entries()][2]).toMatchSnapshot()
        expect([...sepraisal.cubes.keys()]).toMatchSnapshot()
        expect(sepraisal.cubes.size).toEqual(366)
    })
})
