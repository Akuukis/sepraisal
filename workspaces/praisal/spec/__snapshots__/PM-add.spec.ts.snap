// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PraisalManager.addComponents should succeed based on vendor assets 1`] = `
Array [
  "Component/Construction",
  Object {
    "health": 18,
    "mass": 8,
    "maxIntegrity": 30,
    "prerequisites": Object {
      "Ingot/Iron": 8,
    },
    "size": Object {
      "X": 0.2,
      "Y": 0.1,
      "Z": 0.1,
    },
    "subtype": "Construction",
    "time": 1,
    "type": "Component",
    "volume": 2,
  },
]
`;

exports[`PraisalManager.addComponents should succeed based on vendor assets 2`] = `
Array [
  "Component/Construction",
  "Component/Girder",
  "Component/MetalGrid",
  "Component/InteriorPlate",
  "Component/SteelPlate",
  "Component/SmallTube",
  "Component/LargeTube",
  "Component/Motor",
  "Component/Display",
  "Component/BulletproofGlass",
  "Component/Computer",
  "Component/Reactor",
  "Component/Thrust",
  "Component/GravityGenerator",
  "Component/Medical",
  "Component/RadioCommunication",
  "Component/Detector",
  "Component/Canvas",
  "Component/Explosives",
  "Component/SolarCell",
  "Component/PowerCell",
  "Component/Superconductor",
  "Component/ZoneChip",
]
`;

exports[`PraisalManager.addCubes should succeed based on vendor assets 1`] = `
Array [
  "DebugSphere2/DebugSphereLarge",
  Object {
    "$": Object {
      "xsi:type": "MyObjectBuilder_DebugSphere2Definition",
    },
    "BlockTopology": Array [
      "TriangleMesh",
    ],
    "BuildProgressModels": Array [
      Object {
        "Model": Array [
          Object {
            "$": Object {
              "BuildPercentUpperBound": "1.00",
              "File": "Models\\\\Debug\\\\RenderDebugSpheresMetallic.mwm",
            },
          },
        ],
      },
    ],
    "BuildTimeSeconds": Array [
      10,
    ],
    "Components": Array [
      Object {
        "Component": Array [
          Object {
            "$": Object {
              "Count": "10",
              "Subtype": "SteelPlate",
            },
          },
          Object {
            "$": Object {
              "Count": "20",
              "Subtype": "Computer",
            },
          },
        ],
      },
    ],
    "CriticalComponent": Array [
      Object {
        "$": Object {
          "Index": "0",
          "Subtype": "Computer",
        },
      },
    ],
    "CubeSize": Array [
      "Large",
    ],
    "DisplayName": Array [
      "DisplayName_Block_DebugSphere2",
    ],
    "EdgeType": Array [
      "Light",
    ],
    "Icon": Array [
      "Textures\\\\GUI\\\\Icons\\\\Cubes\\\\SpaceBall.dds",
    ],
    "Id": Array [
      Object {
        "SubtypeId": Array [
          "DebugSphereLarge",
        ],
        "TypeId": Array [
          "MyObjectBuilder_DebugSphere2",
        ],
      },
    ],
    "IsAirTight": Array [
      "false",
    ],
    "Model": Array [
      "Models\\\\Debug\\\\RenderDebugSpheresMetallic.mwm",
    ],
    "ModelOffset": Array [
      Object {
        "$": Object {
          "x": "0",
          "y": "0",
          "z": "0",
        },
      },
    ],
    "MountPoints": Array [
      Object {
        "MountPoint": Array [
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "1",
              "Side": "Left",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "1",
              "Side": "Right",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "1",
              "Side": "Back",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "1",
              "Side": "Front",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "2",
              "Side": "Top",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
          Object {
            "$": Object {
              "EndX": "2",
              "EndY": "2",
              "Side": "Bottom",
              "StartX": "0.0",
              "StartY": "0.0",
            },
          },
        ],
      },
    ],
    "PCU": Array [
      0,
    ],
    "Public": Array [
      "false",
    ],
    "RequiredPowerInput": Array [
      "0.0",
    ],
    "Size": Array [
      Object {
        "$": Object {
          "x": 2,
          "y": 1,
          "z": 2,
        },
      },
    ],
  },
]
`;

exports[`PraisalManager.addCubes should succeed based on vendor assets 2`] = `
Array [
  "CubeBlock/LargeRailStraight",
  "DebugSphere1/DebugSphereLarge",
  "DebugSphere2/DebugSphereLarge",
  "DebugSphere3/DebugSphereLarge",
  "CubeBlock/LargeBlockArmorBlock",
  "CubeBlock/LargeBlockArmorSlope",
  "CubeBlock/LargeBlockArmorCorner",
  "CubeBlock/LargeBlockArmorCornerInv",
  "CubeBlock/LargeRoundArmor_Slope",
  "CubeBlock/LargeRoundArmor_Corner",
  "CubeBlock/LargeRoundArmor_CornerInv",
  "CubeBlock/LargeHeavyBlockArmorBlock",
  "CubeBlock/LargeHeavyBlockArmorSlope",
  "CubeBlock/LargeHeavyBlockArmorCorner",
  "CubeBlock/LargeHeavyBlockArmorCornerInv",
  "CubeBlock/SmallBlockArmorBlock",
  "CubeBlock/SmallBlockArmorSlope",
  "CubeBlock/SmallBlockArmorCorner",
  "CubeBlock/SmallBlockArmorCornerInv",
  "CubeBlock/SmallHeavyBlockArmorBlock",
  "CubeBlock/SmallHeavyBlockArmorSlope",
  "CubeBlock/SmallHeavyBlockArmorCorner",
  "CubeBlock/SmallHeavyBlockArmorCornerInv",
  "CubeBlock/LargeHalfArmorBlock",
  "CubeBlock/LargeHeavyHalfArmorBlock",
  "CubeBlock/LargeHalfSlopeArmorBlock",
  "CubeBlock/LargeHeavyHalfSlopeArmorBlock",
  "CubeBlock/HalfArmorBlock",
  "CubeBlock/HeavyHalfArmorBlock",
  "CubeBlock/HalfSlopeArmorBlock",
  "CubeBlock/HeavyHalfSlopeArmorBlock",
  "CubeBlock/LargeBlockArmorRoundSlope",
  "CubeBlock/LargeBlockArmorRoundCorner",
  "CubeBlock/LargeBlockArmorRoundCornerInv",
  "CubeBlock/LargeHeavyBlockArmorRoundSlope",
  "CubeBlock/LargeHeavyBlockArmorRoundCorner",
  "CubeBlock/LargeHeavyBlockArmorRoundCornerInv",
  "CubeBlock/SmallBlockArmorRoundSlope",
  "CubeBlock/SmallBlockArmorRoundCorner",
  "CubeBlock/SmallBlockArmorRoundCornerInv",
  "CubeBlock/SmallHeavyBlockArmorRoundSlope",
  "CubeBlock/SmallHeavyBlockArmorRoundCorner",
  "CubeBlock/SmallHeavyBlockArmorRoundCornerInv",
  "CubeBlock/LargeBlockArmorSlope2Base",
  "CubeBlock/LargeBlockArmorSlope2Tip",
  "CubeBlock/LargeBlockArmorCorner2Base",
  "CubeBlock/LargeBlockArmorCorner2Tip",
  "CubeBlock/LargeBlockArmorInvCorner2Base",
  "CubeBlock/LargeBlockArmorInvCorner2Tip",
  "CubeBlock/LargeHeavyBlockArmorSlope2Base",
  "CubeBlock/LargeHeavyBlockArmorSlope2Tip",
  "CubeBlock/LargeHeavyBlockArmorCorner2Base",
  "CubeBlock/LargeHeavyBlockArmorCorner2Tip",
  "CubeBlock/LargeHeavyBlockArmorInvCorner2Base",
  "CubeBlock/LargeHeavyBlockArmorInvCorner2Tip",
  "CubeBlock/SmallBlockArmorSlope2Base",
  "CubeBlock/SmallBlockArmorSlope2Tip",
  "CubeBlock/SmallBlockArmorCorner2Base",
  "CubeBlock/SmallBlockArmorCorner2Tip",
  "CubeBlock/SmallBlockArmorInvCorner2Base",
  "CubeBlock/SmallBlockArmorInvCorner2Tip",
  "CubeBlock/SmallHeavyBlockArmorSlope2Base",
  "CubeBlock/SmallHeavyBlockArmorSlope2Tip",
  "CubeBlock/SmallHeavyBlockArmorCorner2Base",
  "CubeBlock/SmallHeavyBlockArmorCorner2Tip",
  "CubeBlock/SmallHeavyBlockArmorInvCorner2Base",
  "CubeBlock/SmallHeavyBlockArmorInvCorner2Tip",
  "MyProgrammableBlock/SmallProgrammableBlock",
  "MyObjectBuilder_Projector/LargeProjector",
  "MyObjectBuilder_Projector/SmallProjector",
  "SensorBlock/SmallBlockSensor",
  "SensorBlock/LargeBlockSensor",
  "SoundBlock/SmallBlockSoundBlock",
  "SoundBlock/LargeBlockSoundBlock",
  "ButtonPanel/ButtonPanelLarge",
  "ButtonPanel/ButtonPanelSmall",
  "TimerBlock/TimerBlockLarge",
  "TimerBlock/TimerBlockSmall",
  "MyProgrammableBlock/LargeProgrammableBlock",
  "RadioAntenna/LargeBlockRadioAntenna",
  "Beacon/LargeBlockBeacon",
  "Beacon/SmallBlockBeacon",
  "RadioAntenna/SmallBlockRadioAntenna",
  "RemoteControl/LargeBlockRemoteControl",
  "RemoteControl/SmallBlockRemoteControl",
  "LaserAntenna/LargeBlockLaserAntenna",
  "LaserAntenna/SmallBlockLaserAntenna",
  "TerminalBlock/ControlPanel",
  "TerminalBlock/SmallControlPanel",
  "Cockpit/LargeBlockCockpit",
  "Cockpit/LargeBlockCockpitSeat",
  "Cockpit/SmallBlockCockpit",
  "Cockpit/DBSmallBlockFighterCockpit",
  "Cockpit/CockpitOpen",
  "Gyro/LargeBlockGyro",
  "Gyro/SmallBlockGyro",
  "Cockpit/OpenCockpitSmall",
  "Door/",
  "AirtightHangarDoor/",
  "AirtightSlideDoor/LargeBlockSlideDoor",
  "CubeBlock/ArmorCenter",
  "CubeBlock/ArmorCorner",
  "CubeBlock/ArmorInvCorner",
  "CubeBlock/ArmorSide",
  "CubeBlock/SmallArmorCenter",
  "CubeBlock/SmallArmorCorner",
  "CubeBlock/SmallArmorInvCorner",
  "CubeBlock/SmallArmorSide",
  "BatteryBlock/LargeBlockBatteryBlock",
  "BatteryBlock/SmallBlockBatteryBlock",
  "BatteryBlock/SmallBlockSmallBatteryBlock",
  "Reactor/SmallBlockSmallGenerator",
  "Reactor/SmallBlockLargeGenerator",
  "Reactor/LargeBlockSmallGenerator",
  "Reactor/LargeBlockLargeGenerator",
  "HydrogenEngine/LargeHydrogenEngine",
  "HydrogenEngine/SmallHydrogenEngine",
  "WindTurbine/LargeBlockWindTurbine",
  "SolarPanel/LargeBlockSolarPanel",
  "SolarPanel/SmallBlockSolarPanel",
  "CubeBlock/Monolith",
  "CubeBlock/Stereolith",
  "CubeBlock/LargeDeadAstronaut",
  "GravityGenerator/",
  "GravityGeneratorSphere/",
  "VirtualMass/VirtualMassLarge",
  "VirtualMass/VirtualMassSmall",
  "SpaceBall/SpaceBallLarge",
  "SpaceBall/SpaceBallSmall",
  "Passage/",
  "CubeBlock/LargeStairs",
  "CubeBlock/LargeRamp",
  "CubeBlock/LargeSteelCatwalk",
  "CubeBlock/LargeSteelCatwalk2Sides",
  "CubeBlock/LargeSteelCatwalkCorner",
  "CubeBlock/LargeSteelCatwalkPlate",
  "CubeBlock/LargeCoverWall",
  "CubeBlock/LargeCoverWallHalf",
  "CubeBlock/LargeBlockInteriorWall",
  "CubeBlock/LargeInteriorPillar",
  "Cockpit/PassengerSeatLarge",
  "Cockpit/PassengerSeatSmall",
  "Ladder2/",
  "Ladder2/LadderSmall",
  "TextPanel/SmallTextPanel",
  "TextPanel/SmallLCDPanelWide",
  "TextPanel/SmallLCDPanel",
  "TextPanel/LargeBlockCorner_LCD_1",
  "TextPanel/LargeBlockCorner_LCD_2",
  "TextPanel/LargeBlockCorner_LCD_Flat_1",
  "TextPanel/LargeBlockCorner_LCD_Flat_2",
  "TextPanel/SmallBlockCorner_LCD_1",
  "TextPanel/SmallBlockCorner_LCD_2",
  "TextPanel/SmallBlockCorner_LCD_Flat_1",
  "TextPanel/SmallBlockCorner_LCD_Flat_2",
  "TextPanel/LargeTextPanel",
  "TextPanel/LargeLCDPanel",
  "TextPanel/LargeLCDPanelWide",
  "ReflectorLight/LargeBlockFrontLight",
  "ReflectorLight/SmallBlockFrontLight",
  "InteriorLight/SmallLight",
  "InteriorLight/SmallBlockSmallLight",
  "InteriorLight/LargeBlockLight_1corner",
  "InteriorLight/LargeBlockLight_2corner",
  "InteriorLight/SmallBlockLight_1corner",
  "InteriorLight/SmallBlockLight_2corner",
  "OxygenTank/OxygenTankSmall",
  "OxygenTank/",
  "OxygenTank/LargeHydrogenTank",
  "OxygenTank/SmallHydrogenTank",
  "AirVent/",
  "AirVent/SmallAirVent",
  "CargoContainer/SmallBlockSmallContainer",
  "CargoContainer/SmallBlockMediumContainer",
  "CargoContainer/SmallBlockLargeContainer",
  "CargoContainer/LargeBlockSmallContainer",
  "CargoContainer/LargeBlockLargeContainer",
  "Conveyor/SmallBlockConveyor",
  "Conveyor/LargeBlockConveyor",
  "Collector/Collector",
  "Collector/CollectorSmall",
  "ShipConnector/Connector",
  "ShipConnector/ConnectorSmall",
  "ShipConnector/ConnectorMedium",
  "ConveyorConnector/ConveyorTube",
  "ConveyorConnector/ConveyorTubeSmall",
  "ConveyorConnector/ConveyorTubeMedium",
  "ConveyorConnector/ConveyorFrameMedium",
  "ConveyorConnector/ConveyorTubeCurved",
  "ConveyorConnector/ConveyorTubeSmallCurved",
  "ConveyorConnector/ConveyorTubeCurvedMedium",
  "Conveyor/SmallShipConveyorHub",
  "ConveyorSorter/LargeBlockConveyorSorter",
  "ConveyorSorter/MediumBlockConveyorSorter",
  "ConveyorSorter/SmallBlockConveyorSorter",
  "PistonBase/LargePistonBase",
  "ExtendedPistonBase/LargePistonBase",
  "PistonTop/LargePistonTop",
  "PistonBase/SmallPistonBase",
  "ExtendedPistonBase/SmallPistonBase",
  "PistonTop/SmallPistonTop",
  "MotorStator/LargeStator",
  "MotorRotor/LargeRotor",
  "MotorStator/SmallStator",
  "MotorRotor/SmallRotor",
  "MotorAdvancedStator/LargeAdvancedStator",
  "MotorAdvancedRotor/LargeAdvancedRotor",
  "MotorAdvancedStator/SmallAdvancedStator",
  "MotorAdvancedRotor/SmallAdvancedRotor",
  "MedicalRoom/LargeMedicalRoom",
  "CryoChamber/LargeBlockCryoChamber",
  "CryoChamber/SmallBlockCryoChamber",
  "Refinery/LargeRefinery",
  "Refinery/Blast Furnace",
  "OxygenGenerator/",
  "OxygenGenerator/OxygenGeneratorSmall",
  "Assembler/LargeAssembler",
  "Assembler/BasicAssembler",
  "SurvivalKit/SurvivalKitLarge",
  "SurvivalKit/SurvivalKit",
  "OxygenFarm/LargeBlockOxygenFarm",
  "UpgradeModule/LargeProductivityModule",
  "UpgradeModule/LargeEffectivenessModule",
  "UpgradeModule/LargeEnergyModule",
  "Thrust/SmallBlockSmallThrust",
  "Thrust/SmallBlockLargeThrust",
  "Thrust/LargeBlockSmallThrust",
  "Thrust/LargeBlockLargeThrust",
  "Thrust/LargeBlockLargeHydrogenThrust",
  "Thrust/LargeBlockSmallHydrogenThrust",
  "Thrust/SmallBlockLargeHydrogenThrust",
  "Thrust/SmallBlockSmallHydrogenThrust",
  "Thrust/LargeBlockLargeAtmosphericThrust",
  "Thrust/LargeBlockSmallAtmosphericThrust",
  "Thrust/SmallBlockLargeAtmosphericThrust",
  "Thrust/SmallBlockSmallAtmosphericThrust",
  "Drill/SmallBlockDrill",
  "Drill/LargeBlockDrill",
  "ShipGrinder/LargeShipGrinder",
  "ShipGrinder/SmallShipGrinder",
  "ShipWelder/LargeShipWelder",
  "ShipWelder/SmallShipWelder",
  "OreDetector/LargeOreDetector",
  "OreDetector/SmallBlockOreDetector",
  "LandingGear/LargeBlockLandingGear",
  "LandingGear/SmallBlockLandingGear",
  "JumpDrive/LargeJumpDrive",
  "CameraBlock/SmallCameraBlock",
  "CameraBlock/LargeCameraBlock",
  "MergeBlock/LargeShipMergeBlock",
  "MergeBlock/SmallShipMergeBlock",
  "Parachute/LgParachute",
  "Parachute/SmParachute",
  "Warhead/LargeWarhead",
  "Warhead/SmallWarhead",
  "Decoy/LargeDecoy",
  "Decoy/SmallDecoy",
  "LargeGatlingTurret/",
  "LargeGatlingTurret/SmallGatlingTurret",
  "LargeMissileTurret/",
  "LargeMissileTurret/SmallMissileTurret",
  "InteriorTurret/LargeInteriorTurret",
  "SmallMissileLauncher/",
  "SmallMissileLauncher/LargeMissileLauncher",
  "SmallMissileLauncherReload/SmallRocketLauncherReload",
  "SmallGatlingGun/",
  "MotorSuspension/Suspension3x3",
  "MotorSuspension/Suspension5x5",
  "MotorSuspension/Suspension1x1",
  "MotorSuspension/SmallSuspension3x3",
  "MotorSuspension/SmallSuspension5x5",
  "MotorSuspension/SmallSuspension1x1",
  "MotorSuspension/Suspension3x3mirrored",
  "MotorSuspension/Suspension5x5mirrored",
  "MotorSuspension/Suspension1x1mirrored",
  "MotorSuspension/SmallSuspension3x3mirrored",
  "MotorSuspension/SmallSuspension5x5mirrored",
  "MotorSuspension/SmallSuspension1x1mirrored",
  "Wheel/SmallRealWheel1x1",
  "Wheel/SmallRealWheel",
  "Wheel/SmallRealWheel5x5",
  "Wheel/RealWheel1x1",
  "Wheel/RealWheel",
  "Wheel/RealWheel5x5",
  "Wheel/SmallRealWheel1x1mirrored",
  "Wheel/SmallRealWheelmirrored",
  "Wheel/SmallRealWheel5x5mirrored",
  "Wheel/RealWheel1x1mirrored",
  "Wheel/RealWheelmirrored",
  "Wheel/RealWheel5x5mirrored",
  "Wheel/Wheel1x1",
  "Wheel/SmallWheel1x1",
  "Wheel/Wheel3x3",
  "Wheel/SmallWheel3x3",
  "Wheel/Wheel5x5",
  "Wheel/SmallWheel5x5",
  "CubeBlock/LargeWindowSquare",
  "CubeBlock/LargeWindowEdge",
  "CubeBlock/Window1x2Slope",
  "CubeBlock/Window1x2Inv",
  "CubeBlock/Window1x2Face",
  "CubeBlock/Window1x2SideLeft",
  "CubeBlock/Window1x2SideLeftInv",
  "CubeBlock/Window1x2SideRight",
  "CubeBlock/Window1x2SideRightInv",
  "CubeBlock/Window1x1Slope",
  "CubeBlock/Window1x1Face",
  "CubeBlock/Window1x1Side",
  "CubeBlock/Window1x1SideInv",
  "CubeBlock/Window1x1Inv",
  "CubeBlock/Window1x2Flat",
  "CubeBlock/Window1x2FlatInv",
  "CubeBlock/Window1x1Flat",
  "CubeBlock/Window1x1FlatInv",
  "CubeBlock/Window3x3Flat",
  "CubeBlock/Window3x3FlatInv",
  "CubeBlock/Window2x3Flat",
  "CubeBlock/Window2x3FlatInv",
]
`;

exports[`PraisalManager.addCubes should succeed based on vendor assets 3`] = `318`;

exports[`PraisalManager.addIngots should succeed based on vendor assets 1`] = `
Array [
  "Ingot/Stone",
  Object {
    "mass": 1,
    "prerequisites": Object {
      "Ore/Stone": 71.42857142857143,
    },
    "subtype": "Stone",
    "time": 10,
    "type": "Ingot",
    "volume": 0.37,
  },
]
`;

exports[`PraisalManager.addIngots should succeed based on vendor assets 2`] = `
Array [
  "Ingot/Stone",
  "Ingot/Iron",
  "Ingot/Nickel",
  "Ingot/Cobalt",
  "Ingot/Magnesium",
  "Ingot/Silicon",
  "Ingot/Silver",
  "Ingot/Gold",
  "Ingot/Platinum",
  "Ingot/Uranium",
]
`;

exports[`PraisalManager.addOres should succeed based on vendor assets 1`] = `
Array [
  "Ore/Stone",
  Object {
    "mass": 1,
    "subtype": "Stone",
    "type": "Ore",
    "volume": 0.37,
  },
]
`;

exports[`PraisalManager.addOres should succeed based on vendor assets 2`] = `
Array [
  "Ore/Stone",
  "Ore/Iron",
  "Ore/Nickel",
  "Ore/Cobalt",
  "Ore/Magnesium",
  "Ore/Silicon",
  "Ore/Silver",
  "Ore/Gold",
  "Ore/Platinum",
  "Ore/Uranium",
  "Ore/Scrap",
  "Ore/Ice",
  "Ore/Organic",
]
`;
